pinot-broker/src/test/java/org/apache/pinot/broker/broker/HelixBrokerStarterTest.java

```diff
@@ -18,287 +18,197 @@
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
 
-public class HelixBrokerStarterTest extends ControllerTest {
-  private static final int SEGMENT_COUNT = 6;
-  private static final String RAW_DINING_TABLE_NAME = "dining";
-  private static final String DINING_TABLE_NAME = TableNameBuilder.OFFLINE.tableNameWithType(RAW_DINING_TABLE_NAME);
-  private static final String COFFEE_TABLE_NAME = TableNameBuilder.OFFLINE.tableNameWithType("coffee");
 
-  private final Configuration _brokerConf = new BaseConfiguration();
+public class HelixBrokerStarterTest extends ControllerTest {
+  private static final String RAW_TABLE_NAME = "testTable";
+  private static final String OFFLINE_TABLE_NAME = TableNameBuilder.OFFLINE.tableNameWithType(RAW_TABLE_NAME);
+  private static final String REALTIME_TABLE_NAME = TableNameBuilder.REALTIME.tableNameWithType(RAW_TABLE_NAME);
+  private static final String TIME_COLUMN_NAME = "daysSinceEpoch";
+  private static final int NUM_BROKERS = 3;
+  private static final int NUM_SERVERS = 1;
+  private static final int NUM_OFFLINE_SEGMENTS = 5;
 
-  private ZkClient _zkClient;
-  private HelixBrokerStarter _helixBrokerStarter;
-  private ZkStarter.ZookeeperInstance _zookeeperInstance;
+  private HelixBrokerStarter _brokerStarter;
 
-  @BeforeTest
+  @BeforeClass
   public void setUp()
       throws Exception {
-    _zookeeperInstance = ZkStarter.startLocalZkServer();
-    _zkClient = new ZkClient(ZkStarter.DEFAULT_ZK_STR);
-
+    startZk();
     startController();
 
-    _brokerConf.addProperty(CommonConstants.Helix.KEY_OF_BROKER_QUERY_PORT, 8943);
-    _brokerConf.addProperty(CommonConstants.Broker.CONFIG_OF_BROKER_REFRESH_TIMEBOUNDARY_INFO_SLEEP_INTERVAL, 100L);
-    _helixBrokerStarter = new HelixBrokerStarter(_brokerConf, getHelixClusterName(), ZkStarter.DEFAULT_ZK_STR);
-    _helixBrokerStarter.start();
+    Configuration brokerConf = new BaseConfiguration();
+    brokerConf.addProperty(Helix.KEY_OF_BROKER_QUERY_PORT, 18099);
+    brokerConf.addProperty(Broker.CONFIG_OF_BROKER_REFRESH_TIMEBOUNDARY_INFO_SLEEP_INTERVAL, 100L);
+    _brokerStarter = new HelixBrokerStarter(brokerConf, getHelixClusterName(), ZkStarter.DEFAULT_ZK_STR);
+    _brokerStarter.start();
 
     ControllerRequestBuilderUtil
-        .addFakeBrokerInstancesToAutoJoinHelixCluster(getHelixClusterName(), ZkStarter.DEFAULT_ZK_STR, 5, true);
+        .addFakeBrokerInstancesToAutoJoinHelixCluster(getHelixClusterName(), ZkStarter.DEFAULT_ZK_STR, NUM_BROKERS - 1,
+            true);
     ControllerRequestBuilderUtil
-        .addFakeDataInstancesToAutoJoinHelixCluster(getHelixClusterName(), ZkStarter.DEFAULT_ZK_STR, 1, true);
-
-    while (_helixAdmin.getInstancesInClusterWithTag(getHelixClusterName(), "DefaultTenant_OFFLINE").size() == 0
```

Comment:
Not sure why they had this wait. Does helix take time to recognize the instances added? Is it because of cached vs uncached references for instances in the getInstancesWithTag() call? Make sure that we don't need this wait.

Reply:
This wait is not needed. There are several redundant wait in the old test